rewriter/Data.cc

```diff
@@ -15,30 +15,70 @@ namespace sorbet::rewriter {
 
 namespace {
 
-bool isMissingInitialize(const core::GlobalState &gs, const ast::Send *send) {
+const ast::Send *findParams(ast::ExpressionPtr *send) {
+    ast::Send *sig = ASTUtil::castSig(*send);
+    if (sig == nullptr) {
+        return nullptr;
+    }
+
+    auto *block = sig->block();
+    if (block == nullptr) {
+        return nullptr;
+    }
+
+    auto *bodyBlock = ast::cast_tree<ast::Send>(block->body);
+
+    while (bodyBlock && bodyBlock->fun != core::Names::params()) {
+        bodyBlock = ast::cast_tree<ast::Send>(bodyBlock->recv);
+    }
+
+    return bodyBlock;
+}
+
+optional<pair<ast::MethodDef *, const ast::Send *>> getInitialize(const core::GlobalState &gs, ast::Send *send) {
     if (!send->hasBlock()) {
-        return true;
+        return {};
     }
 
     auto block = send->block();
 
     if (auto *insSeq = ast::cast_tree<ast::InsSeq>(block->body)) {
-        auto methodDef = ast::cast_tree<ast::MethodDef>(insSeq->expr);
+        ast::ExpressionPtr *prevStat = nullptr;
+        for (auto &&stat : insSeq->stats) {
```

Comment:
Is this rvalue reference load bearing here?

The only three other places that we use this `auto &&` pattern in Sorbet, it's because we're going to call `move()` on the loop variable to avoid copying the tree.

In this case, we're not moving trees, so it seems we could drop it.
```suggestion
        for (auto &stat : insSeq->stats) {
```